pages/Documents/ConversationalAI/Third-PartyBots/CustomEndpoint/service-implementation.md

```diff
@@ -0,0 +1,365 @@
+| convId      | LivePerson Conversation Identifier                                                                                            |
+
+Example of the request body of a Text event that is sent by Third-Party
+Bots can be seen in Figure 2.5
+
+```javascript
+{
+  "type": "TEXT",
+  "source": "CONSUMER",
+  "data": {
+    "message": "Hi"
+  },
+  "context": {
+    "lpEvent": {
+      "sequence": 0,
+      "serverTimestamp": 1649331017991,
+      "event": {
+        "type": "ContentEvent",
+        "message": "Hi",
+        "contentType": "text/plain"
+      },
+      "conversationId": "608b2868-f686-4b89-891f-497816958cfc",
+      "dialogId": "608b2868-f686-4b89-891f-497816958cfc",
+      "metadata": [],
+      "messageAudience": "ALL"
+      // .... RAW LP Content Event properties
+    }
+  }
+}
+```
+
+Figure 2.5 Example request body of Text event sent to Send Content Events endpoint
+
+Example of the request body of a Rich Content event that is sent by Third-Party
+Bots can be seen in Figure 2.6
+
+```javascript
+{
+  "type": "RICH_CONTENT",
+  "source": "CONSUMER",
+  "data": {
+    "content": {
+      "type": "vertical",
+      "elements": [
+        {
+          "type": "map",
+          "la": 49.000000000000,
+          "lo": 7.111111111111,
+          "alt": "49.000000000000, 7.111111111111"
+        }
+      ]
+    }
+  },
+  "context": {
+    "lpEvent": {
+      "sequence": 12,
+      "serverTimestamp": 1649332721115,
+      "event": {
+        "type": "RichContentEvent",
+        "content": {
+          "type": "vertical",
+          "elements": [
+            {
+              "type": "map",
+              "la": 49.000000000000,
+              "lo": 7.111111111111,
+              "alt": "49.000000000000, 7.111111111111"
+            }
+          ]
+        }
+      },
+      "conversationId": "608b2868-f686-4b89-891f-497816958cfc",
+      "dialogId": "608b2868-f686-4b89-891f-497816958cfc",
+      "metadata": [],
+      "messageAudience": "ALL"
+      // .... RAW LP Content Event properties
+    }
+  }
+}
+```
+
+Figure 2.6 Example request body of Rich Content event sent to Send Content Events endpoint
+
+### Authorization and Authentication
+
+Third-Party Bots uses [App-JWT OAuth 2.0](accessing-liveperson-apis.html#oauth-20-app-jwt) authentication
+mechanism for a server to server interaction. Third-Party Bots uses the provided
+`Client ID` and `Client Secret` of an App Installation in the vendor configuration to generate bearer tokens.
+More information on the Sentinel API can be found [here](connector-api-send-api-authorization-and-authentication.html#get-appjwt).
+
+Third-Party Bots sends the bearer token inside the `Authorization` header on all requests
+to the Custom Endpoint service. The brands need to ensure the request is authorized, e.g.
+if the account the token has been generated on is allowed to access the addressed bot
+Custom Endpoint service endpoints.
```

Comment:
Maybe.
```suggestion
resources.
```
to make clearer we could be talking just about one bot here in general. Or remove 'bot' if you are trying to imply you are restricting access for the whole service.